src/main/java/appeng/bootstrap/FeatureFactory.java

```diff
@@ -123,39 +127,43 @@ public FeatureFactory features( AEFeature... features )
 		return new FeatureFactory( this, features );
 	}
 
-	void addBootstrapComponent( IBootstrapComponent component )
+	public <T extends IBootstrapComponent> void addBootstrapComponent( T component )
```

Comment:
The generic `T` should not be needed here. We only care about `IBootstrapComponent` here as parameter, but not a concrete subclass/subinterface. This should also avoid the slightly strange `factory.<IItemRegistrationComponent>addBootstrapComponent()` calls.